static/app/views/lastKnownRouteContextProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import {createContext} from 'react';
+
+import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
+import usePrevious from 'sentry/utils/usePrevious';
+import {useRoutes} from 'sentry/utils/useRoutes';
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+export const LastKnownRouteContext = createContext<string>('');
+
+function useLastKnownRoute() {
+  const route = useRoutes();
+  const lastRoute = usePrevious(route);
```

Comment:
I wonder if we should consider excluding the route not found route, since it isn't really "known". Not that it would be a problem I think

Reply:
what do you mean by "exclude"?